### PR TITLE
Remove edit/delete permission for index pages

### PR DESCRIPTION
**Pull request by nathanbegbie**
See https://praekeltorg.atlassian.net/browse/MOLO-99

### CODE REVIEW

File: molo/core/models.py [hunk 1]
```diff
@@ -197,6 +197,11 @@ class SiteSettings(BaseSetting):
     ]
 
 
+class PreventDeleteMixin(object):
+    def hide_delete_button(self):
+        return type(self.get_ancestors().last().specific).__name__ == "Main"
```
Comment (miltontony) [inline molo/core/models.py]: `self.get_parent().depth == 1`

File: molo/core/templatetags/core_tags.py [hunk 2]
```diff
@@ -347,3 +347,11 @@ def social_media_article(context):
         'request': context['request'],
         'locale_code': locale,
     }
+
+
+@register.simple_tag(takes_context=True)
+def should_hide_delete_button(context, page):
+    if hasattr(page, 'hide_delete_button'):
```
Comment (miltontony) [inline molo/core/templatetags/core_tags.py]:
```python
return hasattr(page.specific, 'hide_delete_button')
```

File: molo/core/cookiecutter/scaffold/{{cookiecutter.directory}}/{{cookiecutter.app_name}}/templates/wagtailadmin/pages/edit.html [hunk 3]
```diff
@@ -0,0 +1,119 @@
+{% raw %}
+{% extends "wagtailadmin/pages/edit.html" %}
+{% load wagtailadmin_tags %}
+{% load gravatar %}
+{% load i18n %}
+{% load core_tags %}
+{% block titletag %}{% blocktrans with title=page.get_admin_display_title page_type=content_type.model_class.get_verbose_name %}Editing {{ page_type }}: {{ title }}{% endblocktrans %}{% endblock %}
+{% block bodyclass %}page-editor {% if page.live %}page-is-live{% endif %} model-{{ content_type.model }} {% if page.locked %}page-locked{% endif %}{% endblock %}
+
+{% block content %}
```
Comment (saeedmb8) [inline molo/core/cookiecutter/scaffold/{{cookiecutter.directory}}/{{cookiecutter.app_name}}/templates/wagtailadmin/pages/edit.html]: @nathanbegbie can you please use {{ block.super }} here as well

File: molo/core/cookiecutter/scaffold/{{cookiecutter.directory}}/{{cookiecutter.app_name}}/templates/wagtailadmin/pages/listing/_page_title_explore.html [hunk 4]
```diff
@@ -32,7 +32,14 @@
         <li><a href="{% url 'wagtailadmin_pages:copy' page.id %}" class="button button-small button-secondary">{% trans 'Copy' %}</a></li>
     {% endif %}
     {% if page_perms.can_delete and 'delete' not in hide_actions|default:'' %}
-        <li><a href="{% url 'wagtailadmin_pages:delete' page.id %}" class="button button-small button-secondary">{% trans 'Delete' %}</a></li>
+        {# BEGIN CUSTOM TEMPLATE LOGIC #}
```
Comment (saeedmb8) [inline molo/core/cookiecutter/scaffold/{{cookiecutter.directory}}/{{cookiecutter.app_name}}/templates/wagtailadmin/pages/listing/_page_title_explore.html]: @nathanbegbie would you mind adding a comment around what you add here, so for future will be easier to see what we add to wagtail initial template

File: molo/core/cookiecutter/scaffold/{{cookiecutter.directory}}/{{cookiecutter.app_name}}/templates/wagtailadmin/pages/listing/_button_with_dropdown.html [hunk 5]
```diff
@@ -0,0 +1,24 @@
+{% raw %}
+<div {{ self.attrs }} class="c-dropdown  {% if is_parent %}t-inverted{% else %}t-default{% endif %}" data-dropdown>
+    <a href="javascript:void(0)" title="{{ title }}" class="c-dropdown__button  u-btn-current">
+        {{ label }}
+        <div data-dropdown-toggle class="o-icon  c-dropdown__toggle  [ icon icon-arrow-down ]"></div>
+    </a>
+    <div class="t-dark">
+        <ul role="menu" class="c-dropdown__menu  u-toggle  u-arrow u-arrow--tl u-background">
+        {% for button in buttons %}
```
Comment (saeedmb8) [inline molo/core/cookiecutter/scaffold/{{cookiecutter.directory}}/{{cookiecutter.app_name}}/templates/wagtailadmin/pages/listing/_button_with_dropdown.html]: same here

**Comment by miltontony** [comment]
👍 but I don't like the template overrides - this will make upgrading to future versions of wagtail hella  difficult. But I'm not seeing an easier way at the moment so I'm happy with this.

Also a note for @codiebeulaine with the Merged CMS - We'll need to allow Main to be deletable so we'll have to remove the mixin from Main.

**Pull request merged**